src/common/tusb_fifo.c

```diff
@@ -87,8 +87,15 @@ bool tu_fifo_config(tu_fifo_t *f, void* buffer, uint16_t depth, uint16_t item_si
 
 static inline uint16_t _ff_mod(uint16_t idx, uint16_t depth)
 {
-  while ( idx >= depth) idx -= depth;
-  return idx;
+  //detect power of 2 case
+  if (depth & (depth-1)) {
+    return idx % depth;
```

Comment:
please keep the previous while() loop, the modulo take lots more time than while minus.

Reply:
The idx value is 62 the first time and 65472 on the 9th time. As I said, with the version I debugged the idx only grows until it wraps around. 
the depth is 64.